APIs/schemas/network-address-translation.json

```diff
@@ -18,6 +18,33 @@
       "type": "string",
       "description": "Translation policy name."
     },
+    "translation_scope": {
+      "type": "object",
+      "description": "Indicates where the policy is applied",
+      "required": [
+        "level"
+      ],
+      "properties": {
+        "level": {
+          "description": "Indicates the level at which the policy is applied",
+          "type": "string",
+          "enum": [
+            "root",
+            "network_device",
+            "network_device_interface"
+          ]
+        },
+        "device_id": {
+          "description": "Globally unique identifier of the network device. This property is required if level is 'network_device' or 'network_device_interface', but must not be included if level is 'root'.",
```

Comment:
At the moment I have proposed the relationship between `level` and which other properties are required is covered by the `description` documentation.

Otherwise a more complicated schema is required, which I'm not convinced is worth it. Something like:

```json
"oneOf": [{
  "required": [
    "level"
  ],
  "properties": {
    "level": {
      "description": "Indicates the level at which the policy is applied",
      "type": "string",
      "enum": [
        "root"
      ]
    }
  },
  "additionalProperties": false
},
{
  "required": [
    "level",
    "device_id"
  ],
  "properties": {
    "level": {
      "description": "Indicates the level at which the policy is applied",
      "type": "string",
      "enum": [
        "network_device"
      ]
    },
    "device_id": {
      "description": "Globally unique identifier of the network device. This property is required if level is 'network_device' or 'network_device_interface', but must not be included if level is 'root'.",
    }
  },
  "additionalProperties": false
},
{
  "required": [
    "level",
    "device_id",
    "port_id"
  ],
  "properties": {
    "level": {
      "description": "Indicates the level at which the policy is applied",
      "type": "string",
      "enum": [
        "network_device_interface"
      ]
    },
    "device_id": {
      "description": "Globally unique identifier of the network device. This property is required if level is 'network_device' or 'network_device_interface', but must not be included if level is 'root'.",
    },
    "port_id": {
      "description": "The port identifier of the network device's interface. This property is required if level is 'network_device_interface', but must not be included if level is 'root' or 'network_device'.",
      "type": "string"
    }
  },
  "additionalProperties": false
}]
```

Reply:
@andrewbonney, @vsachin33, thoughts on whether this is a hoop worth jumping through?

Reply:
Provided the spec clearly states that the other options are explicitly disallowed, this is something which could easily be caught by the testing tool. I don't have a strong preference either way.